src/components/cc-ssh-key-list/cc-ssh-key-list.smart.js

```diff
@@ -0,0 +1,145 @@
+import '../cc-smart-container/cc-smart-container.js';
+import { getKeys } from '@clevercloud/client/esm/api/v2/github.js';
+import { get as getUser } from '@clevercloud/client/esm/api/v2/organisation.js';
+import {
+  todo_addSshKey as addSshKey,
+  todo_getSshKeys as getSshKeys,
+  todo_removeSshKey as removeSshKey,
+} from '@clevercloud/client/esm/api/v2/user.js';
+import { ONE_DAY } from '@clevercloud/client/esm/with-cache.js';
+import { defineSmartComponent } from '../../lib/define-smart-component.js';
+import { i18n } from '../../lib/i18n.js';
+import { notifyError, notifySuccess } from '../../lib/notifications.js';
+import { sendToApi } from '../../lib/send-to-api.js';
+import { CcSshKeyList } from './cc-ssh-key-list.js';
+
+defineSmartComponent({
+  selector: 'cc-ssh-key-list',
+  params: {
+    apiConfig: { type: Object },
+  },
+  onContextUpdate ({ component, context, onEvent, updateComponent, signal }) {
+
+    const { apiConfig } = context;
+
+    // Retrieving SSH keys is done in two steps, hidden in the `fetchAllKeys()` implementation:
+    // - first, we retrieve the current user information to check if his/her GitHub account is linked to his/her main account;
+    // - then, we fetch the personal SSH keys and the GitHub keys if needed.
+    // Note: we intentionally show `loading` state only on initial load and not on further actions, to keep a responsive UI.
+    function refreshList () {
+      fetchAllKeys({ apiConfig, signal, cacheDelay: 0 })
+        .then(({ isGithubLinked, personalKeys, githubKeys }) => {
+          updateComponent('keyData', {
+            state: 'loaded',
+            // linked (or unlinked) GitHub account state passed to the component
+            isGithubLinked,
+            // internal key states initialization (to `idle`) after API fetch, to separate fetched data from UI infos
+            personalKeys: personalKeys.map((key) => ({ ...key, state: 'idle' })),
+            githubKeys: githubKeys?.map((key) => ({ ...key, state: 'idle' })),
+          });
+        })
+        .catch((error) => {
+          console.error(error);
+          updateComponent('keyData', { state: 'error' });
+        });
+    }
+
+    onEvent('cc-ssh-key-list:create', ({ name, publicKey }) => {
+      updateComponent('createSshKeyForm', (createSshKeyForm) => {
+        createSshKeyForm.state = 'creating';
+      });
+
+      addKey({ apiConfig, key: { name, key: publicKey } })
+        .then(() => {
+          notifySuccess(component, i18n('cc-ssh-key-list.success.add', { name }));
+          updateComponent('createSshKeyForm', CcSshKeyList.CREATE_FORM_INIT_STATE);
+          refreshList();
```

Comment:
There must be a reason that leads us to refresh the whole list instead of adding the key directly to the model. Can you add a comment to explain why?

Reply:
Yes, there's a good reason :wink: :+1: for a comment